README.rst

```diff
@@ -23,22 +30,304 @@ implementation are in the accepted
+
+    $ git checkout v3.0.2
+
+Then go back up to the root of your repository and copy the ``ah_bootstrap.py``
+file from the submodule to the root of your repository::
+
+    $ cd ..
+    $ cp astropy_helpers/ah_bootstrap.py .
+
+Finally, add::
+
+    import ah_bootstrap
+
+at the top of your ``setup.py`` file. This will ensure that ``astropy_helpers``
+is now available to use in your ``setup.py`` file. Finally, add then commit your
+changes::
+
+    git add astropy_helpers ah_bootstrap.py setup.py
+    git commit ...
+
+Updating astropy-helpers
+------------------------
+
+If you would like to automatically get pull requests to update astropy-helpers,
+then see the instructions `here
+<https://github.com/astropy/astropy-procedures/tree/master/update-affiliated>`_.
+
+To instead update astropy-helpers manually, go inside the submodule and do::
+
+    cd astropy_helpers
+    git fetch origin
+
+Then checkout the version you want to use, e.g.::
+
+    git checkout v3.0.3
+
+Go back up to the root of the repository and update the ``ah_bootstap.py`` file
+too, then add your changes::
+
+    cp astropy_helpers/ah_bootstrap.py .
+    git add astropy_helpers ah_bootstrap.py
+    git commit ...
+
+What does astropy-helpers provide?
+----------------------------------
+
+Custom setup.py commands
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+The main part of astropy-helpers is to provide customized setuptools commands.
+For example, in a package that uses astropy-helpers, the following command
+will be available::
+
+    python setup.py build_docs
+
+and this command is implemented in astropy-helpers. To use the custom
+commands described here, add::
+
+    from astropy_helpers.setup_helpers import register_commands
+
+to your ``setup.py`` file, then do::
+
+    cmdclassd = register_commands(NAME, VERSION, RELEASE)
+
+where ``NAME`` is the name of your package, ``VERSION`` is the full version
+string, and ``RELEASE`` is a boolean value indicating whether the version is
+a stable released version (``True``) or a developer version (``False``).
+Finally, pass ``cmdclassd`` to the ``setup`` function::
+
+     setup(...,
+           cmdclass=cmdclassd)
+
+The commands we provide or customize are:
+
+**python setup.py test**
+
+This command will automatically build the package, install it to a temporary
+directory, and run the tests using `pytest <http://pytest.org/>`_ on this
+installed version. Note that the bulk of this command is actually defined
+in ``astropy.tests.command.AstropyTest`` - here we simply define the custom
+setuptools command.
+
+**python setup.py sdist**
+
+We redefine ``sdist`` to use the version from distutils rather than from
+setuptools, as the setuptools version requires duplication of information
+in ``MANIFEST.in``.
+
+**python setup.py build_docs**
+
+This command will automatically build the package, then run sphinx to build
+the documentation. Sphinx itself provides a custom setuptools command, which we
```

Comment:
```suggestion
the documentation. This makes development much easier because it ensures
sphinx extensions that use the package's  code to make documentation are 
actually using the in-development version of the code. Sphinx itself
provides a custom setuptools command, which we
```